lib/sisimai/bite/json/sendgrid.rb

```diff
@@ -0,0 +1,79 @@
+module Sisimai::Bite::JSON
+  # Sisimai::Bite::JSON::SendGrid parses a bounce object(JSON) which created by
+  # SendGrid. Methods in the module are called from only Sisimai::Message.
+  module SendGrid
+    # Imported from p5-Sisimail/lib/Sisimai/Bite/JSON/SendGrid.pm
+    class << self
+      require 'sisimai/bite/json'
+
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def description; return 'SendGrid(JSON): http://sendgrid.com/'; end
+
+      # @abstract Adapt SendGrid bounce object for Sisimai::Message format
+      # @param        [Hash] argvs  bounce object(JSON) retrieved from SendGrid API
+      # @return       [Hash, Nil]   Bounce data list and message/rfc822 part
+      #                             or Undef if it failed to parse or the
+      #                             arguments are missing
+      # @since v4.20.0
+      def adapt(argvs)
+        return nil unless argvs.is_a? Hash
+        return nil unless argvs.keys.size > 0
+        return nil unless argvs.key?('email')
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for adapt is too high. [36.19/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->